.github/docs/README.md

```diff
@@ -13,6 +13,65 @@ This documentation is a work in progress and will be updated as new features are
 | **BaseFeePerGas** | The minimum gas price required for a transaction to be included in the current block.|
 
 
+## Methodology
+
+### Before Galactica
+
+The pre-Galactica transaction model implements a fixed gas pricing mechanism with the following characteristics:
+
+1. **Limited Gas Price Range**: Legacy transactions operate within a constrained gas price range
+2. **Fixed Validator Tip**: Validators receive a fixed 30% of the total gas fee paid by users
+3. **Base Gas Price**: A network parameter (currently set at 1e13 wei) that can be adjusted by the steering committee
+
+### After Galactica
+
+Galactica introduces a more flexible and efficient fee mechanism that supports two transaction types:
+
+#### 1. Legacy Transactions
+- Maintains backward compatibility with the original gas price formula:
+  ```
+  gasPrice = BaseGasPrice * (1 + GasPriceCoef/255)
+  ```
+- Continues to operate within the established gas price range
+
+#### 2. Dynamic Fee Transactions
+- Introduces two new parameters:
+  - `maxFeePerGas`: Maximum total fee per gas unit
+  - `maxPriorityFeePerGas`: Maximum priority fee (tip) per gas unit
+- Both parameters can range from 0 to 2^256 - 1 wei, offering significantly greater flexibility
+- Implements EIP-1559 style dynamic fee market mechanism
+
+#### Fee Market Mechanism
+
+1. **Base Fee Adjustment**:
+   - Automatically adjusts based on previous block's gas utilization
+   - Base fee is completely burned, reducing overall token supply
+
+2. **Transaction Field Mapping**:
+   - Both transaction types use `maxFeePerGas` and `maxPriorityFeePerGas`
+   - For legacy transactions:
+     ```
+     maxFeePerGas = gasPrice
+     maxPriorityFeePerGas = gasPrice
+     ```
+
+3. **Effective Fee Calculation**:
+   - Effective Gas Price (amount user pays per gas):
+     ```
+     Min(block.BaseFee + tx.maxPriorityFeePerGas, tx.maxFeePerGas)
+     ```
+   - Effective Priority Fee (tip to validator per gas):
+     ```
+     Min(tx.maxPriorityFeePerGas, tx.maxFeePerGas - block.BaseFee)
+     ```
+#### Notable Changes
+
+1. **Legacy Transaction Gas Price Constraints**: Legacy transactions are bound by a fixed gas price range of `1e13` to `2 * 1e13 wei`. These transactions may become un-executable during high network activity periods when `block.baseFee` exceeds `2 * 1e13 wei`.
+
+2. **Base Fee Refund Mechanism**: Dynamic fee transactions introduce a refund feature through independent `maxFeePerGas` and `maxPriorityFeePerGas` settings. When `block.baseFee` is less than `tx.maxFeePerGas - tx.maxPriorityFeePerGas`, users receive a refund of `tx.maxFeePerGas - tx.maxPriorityFeePerGas - block.baseFee` per gas unit. Legacy transactions do not support this refund mechanism.
```

Comment:
I'm not really sure about this section

> users receive a refund of `tx.maxFeePerGas - tx.maxPriorityFeePerGas - block.baseFee`

ex:
- baseFee: 100
- priority:     10
- maxFee:  210

effective gas price: min(100+10, 210) -> 110

refund formula: `tx.maxFeePerGas - tx.maxPriorityFeePerGas - block.baseFee` -> 210-10-100=100
effective refund:                                                                                                            -> 110

Reply:
I might get what you mean by 'effective refund: -> 110', which is when you do refund after tx executed, which means `effectiveGasPrice * gasLeftOver`. This is to refund the gas that not used.

Which is not I mean in the statement.  When a user sending a tx, they specified a `maxFeePerGas` means they are willing to pay `maxFeePerGas * gas` for the tx, when block base fee drops down, they don't need to pay that much, and we charged less, `maxFeePerGas-maxPriorityFeePerGas - block.baseFee` is the price they have got for a refund, comparing to their initial thought.

Reply:
Ah, ok, I get what you mean. I was confused because I get the `tx.maxFeePerGas - tx.maxPriorityFeePerGas - block.baseFee` formula, it's the max available minus what the user pays, but it's not really the gasPrice that is used to calculate the refund